frameworks/hdfs/tools/install-secure-hdfs.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+SPARK_BUILD_DIR="${DIR}/../../"
+S3_PATH=""
+HDFS_PRIMARY="hdfs"
+HTTP_PRIMARY="HTTP"
+FRAMEWORK_NAME="hdfs"
+DOMAIN="${FRAMEWORK_NAME}.autoip.dcos.thisdcos.directory"
+REALM="LOCAL"
+LINUX_USER="core"
+KEYTAB_FILE="hdfs.keytab"
+
+function add_kdc() {
+    echo "Adding kdc marathon app..."
+    # dcos marathon app add kdc.json
+
+    echo "Waiting for app to run..."
+    while true; do
+        local TASKS_RUNNING=$(dcos marathon app list --json | jq ".[0].tasksRunning")
+        if [[ ${TASKS_RUNNING} -eq 1 ]]; then
+            break
+        fi
+    done
+
+    sleep 10
+
+    SLAVE_ID=$(dcos task --json | jq -r ".[0].slave_id")
+    MASTER_PUBLIC_IP=$(curl --header "Authorization: token=$(dcos config show core.dcos_acs_token)" $(dcos config show core.dcos_url)/metadata | jq -r ".PUBLIC_IPV4")
+    SLAVE_HOSTNAME=$(dcos node --json | jq -r ".[] | select(.id==\"${SLAVE_ID}\") | .hostname")
+
+    echo "Getting docker container id..."
+    DOCKER_PS_CMD="docker ps | sed -n '2p' | cut -d\" \" -f1"
+    DOCKER_CONTAINER_ID=$(dcos node ssh --master-proxy --mesos-id=${SLAVE_ID} --option StrictHostKeyChecking=no "${DOCKER_PS_CMD}")
+    echo "DOCKER_CONTAINER_ID=${DOCKER_CONTAINER_ID}"
```

Comment:
This will not work the first time...
Connection to 35.163.112.137 closed.
+ DOCKER_CONTAINER_ID=CONTAINER
+ echo DOCKER_CONTAINER_ID=CONTAINER
DOCKER_CONTAINER_ID=CONTAINER
+ add_kerberos_principals
+ echo 'Adding Kerberos principals...'
Adding Kerberos principals...
+ dcos node ssh --master-proxy --mesos-id=0d99ae8d-864f-4dd4-98fb-54c6ec7069ae-S0 --option StrictHostKeyChecking=no 'docker exec CONTAINER kadmin -l add --use-defaults --random-password hdfs/name-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL'
Running `ssh -A -t -o StrictHostKeyChecking=no core@35.163.112.137 ssh -A -t -o StrictHostKeyChecking=no core@10.0.0.15 docker exec CONTAINER kadmin -l add --use-defaults --random-password hdfs/name-0-node.hdfs.autoip.dcos.thisdcos.directory@LOCAL`
Error response from daemon: No such container: CONTAINER

Had to re-run it probably the sleep time is not enough...